src/lib_main.rs

```diff
@@ -0,0 +1,327 @@
+
+    #[cfg(all(target_os = "windows", feature = "gui"))]
+    if CFG.set(kanata_arc.clone()).is_err() {
+        warn!("Someone else set our ‘CFG’");
+    }; // store a clone of cfg so that we can ask it to reset itself
+
+    if !args.nodelay {
+        info!("Sleeping for 2s. Please release all keys and don't press additional ones. Run kanata with --help to see how understand more and how to disable this sleep.");
+        std::thread::sleep(std::time::Duration::from_secs(2));
+    }
+
+    // Start a processing loop in another thread and run the event loop in this thread.
+    //
+    // The reason for two different event loops is that the "event loop" only listens for keyboard
+    // events, which it sends to the "processing loop". The processing loop handles keyboard events
+    // while also maintaining `tick()` calls to keyberon.
+
+    let (tx, rx) = std::sync::mpsc::sync_channel(100);
+
+    let (server, ntx, nrx) = if let Some(address) = {
+        #[cfg(feature = "tcp_server")]
+        {
+            args.tcp_server_address
+        }
+        #[cfg(not(feature = "tcp_server"))]
+        {
+            None::<SocketAddrWrapper>
+        }
+    } {
+        let mut server = TcpServer::new(address.into_inner(), tx.clone());
+        server.start(kanata_arc.clone());
+        let (ntx, nrx) = std::sync::mpsc::sync_channel(100);
+        (Some(server), Some(ntx), Some(nrx))
+    } else {
+        (None, None, None)
+    };
+
+    #[cfg(any(not(target_os = "windows"), not(feature = "gui")))]
+    {
+        Kanata::start_processing_loop(kanata_arc.clone(), rx, ntx, args.nodelay);
+
+        if let (Some(server), Some(nrx)) = (server, nrx) {
+            #[allow(clippy::unit_arg)]
+            Kanata::start_notification_loop(nrx, server.connections);
+        }
+
+        #[cfg(target_os = "linux")]
+        sd_notify::notify(true, &[sd_notify::NotifyState::Ready])?;
+
+        Kanata::event_loop(kanata_arc, tx)?;
+    }
+
+    #[cfg(all(target_os = "windows", feature = "gui"))]
+    {
+        use anyhow::Context;
+
+        native_windows_gui::init().context("Failed to init Native Windows GUI")?;
+        let ui = build_tray(&kanata_arc)?;
+        let gui_tx = ui.layer_notice.sender();
+        if GUI_TX.set(gui_tx).is_err() {
+            warn!("Someone else set our ‘GUI_TX’");
+        };
+        Kanata::start_processing_loop(kanata_arc.clone(), rx, ntx, args.nodelay);
+
+        if let (Some(server), Some(nrx)) = (server, nrx) {
+            #[allow(clippy::unit_arg)]
+            Kanata::start_notification_loop(nrx, server.connections);
+        }
+
+        Kanata::event_loop(kanata_arc, tx, ui)?;
+    }
+
+    Ok(())
+}
+pub fn lib_main_cli() -> Result<()> {
+    let ret = main_impl();
+    if let Err(ref e) = ret {
+        log::error!("{e}\n");
+    }
+    eprintln!("\nPress enter to exit");
+    let _ = std::io::stdin().read_line(&mut String::new());
+    ret
+}
+#[cfg(all(target_os = "windows", feature = "gui"))]
```

Comment:
All the code below should be moved to `gui_win.rs`

Reply:
moved